### ISSUE TITLE
Can't build example app

### ISSUE THREAD

**Issue by kolyshkin** [OWASP/Amass#300]
Taking an example from https://github.com/OWASP/Amass/blob/c9f40430e379aa94673013d92aaf95eb04aa1fbb/examples/minimal.go and building it out of tree is impossible due to mixed semver/non-semver dependencies. Please find complete reproducers below.

### Try 1: use go run

```console
$ mkdir try1 && cd try1
$ wget https://raw.githubusercontent.com/OWASP/Amass/c9f40430e379aa94673013d92aaf95eb04aa1fbb/examples/minimal.go
$ export GO111MODULE=on
$ go run minimal.go 
build command-line-arguments: cannot load github.com/OWASP/Amass/config: module github.com/OWASP/Amass@latest found (v2.6.0+incompatible), but does not contain package github.com/OWASP/Amass/config
```

NOTE: I am puzzled as to why v2.6.0 is used here, maybe it's because this is the first version that added `go.mod`?

NOTE: when looking at the tags for this package, I see a mix of ones in the form of `X.Y.Z` and `vX.Y.Z`.

### Try 2: use go mod

Using the above example as a base, let's try to add `go.mod` file:

```console
$ go mod init ex.am/ple
go: creating new go.mod: module ex.am/ple
$ go mod tidy
ex.am/ple imports
	github.com/OWASP/Amass/config: module github.com/OWASP/Amass@latest found (v2.6.0+incompatible), but does not contain package github.com/OWASP/Amass/config
ex.am/ple imports
	github.com/OWASP/Amass/enum: module github.com/OWASP/Amass@latest found (v2.6.0+incompatible), but does not contain package github.com/OWASP/Amass/enum
ex.am/ple imports
	github.com/OWASP/Amass/services: module github.com/OWASP/Amass@latest found (v2.6.0+incompatible), but does not contain package github.com/OWASP/Amass/services
```

### Try 3: pin version

Using the above example as a base, let's try to pin version to v3.2.1:

```console
$ cat go.mod 
module ex.am/ple

go 1.13
$ go mod edit -require github.com/OWASP/Amass@v3.2.1
$ cat go.mod
module ex.am/ple

go 1.13

require github.com/OWASP/Amass v3.2.1
$ go mod tidy
go: finding github.com/OWASP/Amass v3.2.1
go: finding github.com/OWASP/Amass v3.2.1
go: errors parsing go.mod:
/home/kir/tmp/try1/go.mod:5: require github.com/OWASP/Amass: version "v3.2.1" invalid: module contains a go.mod file, so major version must be compatible: should be v0 or v1, not v3
```

NOTE: I believe it happens because this package is not conformant to semver requirements (see more detailed description and the proposed fix at https://github.com/OWASP/Amass/pull/299).

### Try 4: abandon `go.mod` and try using `vendor/` subdir.

From a clean slate:

```console
$ export GO111MODULE=off
$ cd $GOPATH
$ mkdir -p ex.am/ple && cd ex.am/ple
$ wget https://raw.githubusercontent.com/OWASP/Amass/c9f40430e379aa94673013d92aaf95eb04aa1fbb/examples/minimal.go
$ export GO111MODULE=off
$ # FIXME: provide instructions to populate vendor/ subdir
$ go build .
vendor/github.com/prometheus/client_golang/prometheus/desc.go:22:2: cannot find package "github.com/cespare/xxhash/v2" in any of:
	${GOPATH}/vendor/github.com/cespare/xxhash/v2 (vendor tree)
	/snap/go/4668/src/github.com/cespare/xxhash/v2 (from $GOROOT)
	${GOPATH}/src/github.com/cespare/xxhash/v2 (from $GOPATH)
```

This fails because https://github.com/cespare/xxhash is already using v2 semver suffix and old vendoring tools can't work with it.

**Comment by kolyshkin** [comment]
Fixed by #299

**Comment by kolyshkin** [comment]
Once a release is tagged, that is
